pkg/controller/uniregistrysyncer/uniregistrysyncer.go

```diff
@@ -0,0 +1,275 @@
+	); err != nil {
+		return fmt.Errorf("failed to create watch for ImageStreams: %w", err)
+	}
+
+	r.log.Info("Successfully added reconciler to manager")
+	return nil
+}
+
+type objectFilter func(types.NamespacedName) bool
+
+// handlerFactory produces a handler that:
+// * Watches ImageStreams because ImageStreamTags do not support the watch verb
+// * Extracts all ImageStramTags out of the Image
+// * Filters out the ones that are not in use
+// Note: We can not use a predicate because that is directly applied on the source and the source yields ImageStreams, not ImageStreamTags
+// * Creates a reconcile.Request per cluster and ImageStreamTag
+func handlerFactory(filter objectFilter) handler.EventHandler {
+	return imagestreamtagmapper.New(func(in reconcile.Request) []reconcile.Request {
+		if !filter(in.NamespacedName) {
+			return nil
+		}
+		return []reconcile.Request{in}
+	})
+}
+
+type reconciler struct {
+	log         *logrus.Entry
+	appCIClient ctrlruntimeclient.Client
+	apiCIClient ctrlruntimeclient.Client
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithField("request", req.String())
+	log.Info("Starting reconciliation")
+	err := r.reconcile(ctx, req, log)
+	// Ignore the logging for IsConflict errors because they are results of concurrent reconciling
+	if err != nil && !apierrors.IsConflict(err) && !apierrors.IsAlreadyExists(err) {
+		log.WithError(err).Error("Reconciliation failed")
+	} else {
+		log.Info("Finished reconciliation")
+	}
+	return reconcile.Result{}, controllerutil.SwallowIfTerminal(err)
+}
+
+const (
+	annotationDPTPRequester = "dptp.openshift.io/requester"
+)
+
+func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *logrus.Entry) error {
+	sourceImageStreamTag := &imagev1.ImageStreamTag{}
+	sourceClient := r.appCIClient
+	if err := sourceClient.Get(ctx, req.NamespacedName, sourceImageStreamTag); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debug("the source imageStreamTag not found")
+			return nil
+		}
+		return fmt.Errorf("failed to get source imageStreamTag %s: %w", req.NamespacedName.String(), err)
+	}
+
+	imageStreamNameAndTag := strings.Split(req.Name, ":")
+	if n := len(imageStreamNameAndTag); n != 2 {
+		return fmt.Errorf("when splitting imagestreamtagname %s by : expected two results, got %d", req.Name, n)
+	}
+	imageStreamName, imageTag := imageStreamNameAndTag[0], imageStreamNameAndTag[1]
+	isName := types.NamespacedName{Namespace: req.Namespace, Name: imageStreamName}
+	sourceImageStream := &imagev1.ImageStream{}
+	if err := sourceClient.Get(ctx, isName, sourceImageStream); err != nil {
+		// received a request on the isTag, but the 'is' is no longer there
+		return fmt.Errorf("failed to get source imageStream %s: %w", isName.String(), err)
+	}
+
+	*log = *log.WithField("docker_image_reference", sourceImageStreamTag.Image.DockerImageReference)
+
+	targetClient := r.apiCIClient
```

Comment:
same here, why the re-assignment of the client under a new name? It obfuscates the code